cmake/ConfigureCompiler.cmake

```
@@ -89,16 +89,33 @@ else()
     set(GCC YES)
   endif()
 
+  # Use the linker environmental variable, if specified and valid
+  if ((USE_LD STREQUAL "DEFAULT") AND (NOT "$ENV{USE_LD}" STREQUAL ""))
+    string(TOUPPER "$ENV{USE_LD}" USE_LDENV)
+    if (("${USE_LDENV}" STREQUAL "LD") OR ("${USE_LDENV}" STREQUAL "GOLD") OR ("${USE_LDENV}" STREQUAL "LLD") OR ("${USE_LDENV}" STREQUAL "BFD"))
+      set(USE_LD "${USE_LDENV}")
+    endif()
+  endif()
+
   # check linker flags.
-  if ((NOT (USE_LD STREQUAL "LD")) AND (NOT (USE_LD STREQUAL "GOLD")) AND (NOT (USE_LD STREQUAL "LLD")))
-    message (FATAL_ERROR "USE_LD must be set to LD, GOLD, or LLD!")
+  if (USE_LD STREQUAL "DEFAULT")
+    set(USE_LD "LD")
+	else()
+    if ((NOT (USE_LD STREQUAL "LD")) AND (NOT (USE_LD STREQUAL "GOLD")) AND (NOT (USE_LD STREQUAL "LLD")) AND (NOT (USE_LD STREQUAL "BFD")))
+      message (FATAL_ERROR "USE_LD must be set to LD, GOLD, or LLD!")
```

```suggestion
      message (FATAL_ERROR "USE_LD must be set to DEFAULT, LD, GOLD, or LLD!")
```
I think this should list all possible options